### PR TITLE
feat: add cli flag to disable light client server

### LINKED ISSUE #6800: Add cli flag to disable light client server
### Problem description

There's no flag to disable the light client server. It's not critical functionality and it could easily be disabled with no problems for a beacon node.

### Solution description

Add a flag to disable the light client server, eg: `--disableLightClientServer`

### Additional context

Came up during electra interop

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Disable the lightclient server in situations where needed. 

**Description**

- Disable lightclient server 
- Don't enable the reqresp protocols for lightclient
- Don't server lightclient REST endpoints. 

Closes #6800

**Steps to test or reproduce**

Run all tests

### CODE REVIEW

File: packages/beacon-node/src/api/impl/lightclient/index.ts [hunk 1]
```diff
@@ -12,42 +12,55 @@ export function getLightclientApi({
 }: Pick<ApiModules, "chain" | "config">): ApplicationMethods<routes.lightclient.Endpoints> {
   return {
     async getLightClientUpdatesByRange({startPeriod, count}) {
+      const lightClientServer = chain.lightClientServer;
```
Comment (nflaig) [inline packages/beacon-node/src/api/impl/lightclient/index.ts:15]: could just make this assignment before the `return`, and then just do the assert at the begging of the function

File: packages/beacon-node/src/node/utils/lightclient.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import {LightClientServer} from "../../chain/lightClient/index.js";
+
+export function assertLightClientServer(server: LightClientServer | undefined): asserts server is LightClientServer {
+  if (!server) {
+    throw Error("LightclientServer is not running on this node. To server this endpoint we need this");
```
Comment (nflaig) [inline packages/beacon-node/src/node/utils/lightclient.ts]: This is an error message we return on the api, can make it more succinct

```suggestion
    throw Error("Light client server is disabled");
```

File: packages/cli/src/cmds/beacon/options.ts [hunk 3]
```diff
@@ -140,6 +141,11 @@ export const beaconExtraOptions: CliCommandOptions<BeaconExtraArgs> = {
     description: "Attach the beacon node to `globalThis`. Useful to inspect a running beacon node.",
     type: "boolean",
   },
+
+  disableLightClientServer: {
+    description: "Disable lightclient server.",
```
Comment (nflaig) [inline packages/cli/src/cmds/beacon/options.ts]: for everything user facing in text we might wanna use "light client"

```suggestion
    description: "Disable light client server.",
```

File: packages/beacon-node/src/api/impl/lightclient/index.ts [hunk 4]
```diff
@@ -12,42 +12,55 @@ export function getLightclientApi({
...
       return {
         data: updates,
         meta: {versions: updates.map((update) => config.getForkName(update.attestedHeader.beacon.slot))},
       };
     },
 
     async getLightClientOptimisticUpdate() {
-      const update = chain.lightClientServer.getOptimisticUpdate();
+      const lightClientServer = chain.lightClientServer;
+      assertLightClientServer(lightClientServer);
```
Comment (nflaig) [inline packages/beacon-node/src/api/impl/lightclient/index.ts]: re-assignment here is not needed, type assertion works either way

```suggestion
      assertLightClientServer(chain.lightClientServer);
```

**Comment by wemeetagain** [comment]
We should also disable light client gossip topics

File: packages/beacon-node/src/api/impl/lightclient/index.ts (hunk 4 above)
Comment (nazarhussain) [reply packages/beacon-node/src/api/impl/lightclient/index.ts]: Nested attributes type assertion does not work.

> 'chain.lightClientServer' is possibly 'undefined'

Comment (nflaig) [reply packages/beacon-node/src/api/impl/lightclient/index.ts]: it only does not work if you use it in a callback, but here it works, in the other function above it does not

File: packages/beacon-node/src/network/reqresp/handlers/lightClientBootstrap.ts [hunk 5]
```diff
@@ -8,10 +8,14 @@ import {
 import {Root} from "@lodestar/types";
 import {IBeaconChain} from "../../../chain/index.js";
 import {ReqRespMethod, responseSszTypeByMethod} from "../types.js";
+import {assertLightClientServer} from "../../../node/utils/lightclient.js";
 
 export async function* onLightClientBootstrap(requestBody: Root, chain: IBeaconChain): AsyncIterable<ResponseOutgoing> {
+  const lightClientServer = chain.lightClientServer;
+  assertLightClientServer(lightClientServer);
```
Comment (nflaig) [inline packages/beacon-node/src/network/reqresp/handlers/lightClientBootstrap.ts]: could clean up more of those reassignments here

**Review by nflaig — COMMENTED** [review]
LGTM - just minor nit regarding re-assignments

File: packages/beacon-node/src/network/reqresp/handlers/lightClientBootstrap.ts (hunk 5 above)
Comment (nflaig) [reply packages/beacon-node/src/network/reqresp/handlers/lightClientBootstrap.ts]: there are a bunch more, I haven't commented each line expliclty

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.21.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.21.0) :tada:

**Pull request merged**